### ISSUE TITLE
Embedding not reproducible? [token_representation] giving different results each time

### ISSUE THREAD

**Issue by gaceladri** [facebookresearch/esm#53]
Hello,

I was trying to get the embedding representations from the MSA transformer.
The code:

```
import torch
import esm

# Load ESM-1b model
model, alphabet = esm.pretrained.esm_msa1_t12_100M_UR50S()
batch_converter = alphabet.get_batch_converter()

# Prepare data (first 2 sequences from ESMStructuralSplitDataset superfamily / 4)
data = [
    ("protein1", "MKTVRQERLKSIVRILERSKEPVSGAQLAEELSVSRQVIVQDIAYLRSLGYNIVATPRGYVLAGG"),
    ("protein2", "MKTVRQERLKSIVRILERSKEPVSGAQLAEELSVSRQVIVQDIAYLRSLGYIIVATPRGYVLAGG"),
]
batch_labels, batch_strs, batch_tokens = batch_converter(data)

# Extract per-residue representations (on CPU)
with torch.no_grad():
    results = model(batch_tokens, repr_layers=[12], return_contacts=True)
token_representations = results["representations"][12]
```

Each time that I run the code, token_representations is different. Is this the expected behavior?

**Comment by tomsercu** [comment]
Ah thanks for calling that out! The models have some dropout in them, so to make them non-stochastic you should change the model to `eval()` mode; call `model.eval()` after loading it. This is happening in the extract.py script but the README quickstart example could also do that.

**Comment by gaceladri** [comment]
Thanks, I will do that.

**Comment by gaceladri** [comment]
It worked fine. If you want to fix the readme before closing... If not we can close the issue.

**Comment by tomsercu** [comment]
let's leave README as-is, extracting features with dropout can make sense. Definitely for small datasets, can add some extra regularization, why not :)

**Comment by pzhang84** [comment]
I understand that the model was trained on MSAs with an average depth of 1192. In the provided example with only two sequences, how does the MSA Transformer adapt? Is it designed to handle varying sequence numbers (single sequences or many sequences) effectively during inference/embedding? @tomsercu @gaceladri  

Thank you in advance for your help!
